src/expire.c

```diff
@@ -591,12 +596,29 @@ void ttlGenericCommand(client *c, int output_ms) {
 
 /* TTL key */
 void ttlCommand(client *c) {
-    ttlGenericCommand(c, 0);
+    ttlGenericCommand(c, 0, 0);
 }
 
 /* PTTL key */
 void pttlCommand(client *c) {
-    ttlGenericCommand(c, 1);
+    ttlGenericCommand(c, 1, 0);
+}
+
+/* EXPIRETIME key TIME|PTIME */
```

Comment:
I see.

I was under the impression that OSS Redis prefers to limit the number of new commands, meaning if two commands essentially perform the same functionality, then we prefer to combine them.

`EXPIRETIME` and `PEXPIRETIME` are two commands that would be performing very similar functionalities. Are we sure we want to have two separate commands?

Btw, I think you meant to say `EXPIRETIME` should just always return seconds, since that is the unit `TTL` and `EXPIREAT` deal with.

I don't feel strongly either way. If we decide to implement two separate commands. I don't see why not implement both now.

Reply:
yes. we do wanna keep the number of commands low, and have commands that do similar things unified into one.
the main reason for that is that if we have two variants with some boolean flag difference between them (e.g. SET and SETNX), when we add another mutation (SETGET) we can end up with 4 commands, and if we add another one (SETKEEPTTL) we have 8.

so in that spirit, maybe one option is to deprecate `PTTL`, and change `TTL` to have `ABS` and `MS` arguments.
that would mean that TTL can never be made variadic.
for some reason i think that in this case i rather add `PEXPIRETIME` (and possibly `EXPIRETIME`, not a must) which will take no arguments and can later be made variadic.

i'm still not certain what's best here.
@yossigo @madolson WDYT?

Reply:
My first choice was to extend TTL to have ```ABS``` and ```MS``` arguments, and deprecate the PTTL command. They all logically fit together, and I do think they are more of debugging commands and less for production workloads. I don't think there is much value in making the TTL command variadic, I can't think of a very compelling case for it. This issue of variadic commands keeps coming up though, so we probably want to figure out the best syntax for having flags + multiple key names.

My second choice was the current implementation, PEXPIRETIME and EXPIRETIME. I don't think expire time is a must, but it seems straightforward and easy enough to add.

Reply:
I will defer to you folks making this decision. But I will leave my 2 cents here: `TTL ABS` is a bad choice. It's contradictory within itself. 

`TTL` means time-to-live. It's inherently a relative-time-based concept. The first sentence of its [current API doc](https://redis.io/commands/ttl) says:
```
Returns the remaining time to live of a key that has a timeout.
```

Adding a `ABS` option simply invalidates that sentence, and complicates the core purpose of this API.

Reply:
you're right. 
In theory we can resolve that by making the new all purpose command named EXPIRETIME (will take ABS/TTL/MS arguments).
 
But maybe the most straight forward thing to do is add just a PEXPIRETIME command with no arguments and skip EXPIRETIME.

Reply:
I agree `TTL ABS` is confusing. I think having both `PEXPIRETIME` and `EXPIRETIME` makes sense:

* It's only a single extra command, not a matrix like in other cases
* More consistent with existing expiration family of commands
* Can be variadic (although I agree the value is not terribly high)

Reply:
Implemented `PEXPIRETIME` and `EXPIRETIME`